shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptPredicateColumnTokenGenerator.java

```diff
@@ -75,38 +78,54 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
     private Collection<SubstitutableColumnNameToken> generateSQLTokens(final Collection<ExpressionSegment> predicates, final Map<String, String> columnTableNames) {
         Collection<SubstitutableColumnNameToken> result = new LinkedList<>();
         for (ExpressionSegment each : predicates) {
-            Optional<ColumnSegment> column = ColumnExtractor.extract(each);
-            if (!column.isPresent()) {
+            result.addAll(processColumnSegment(ColumnExtractor.extractAll(each), columnTableNames));
+        }
+        return result;
+    }
+    
+    private List<SubstitutableColumnNameToken> processColumnSegment(final Collection<Optional<ColumnSegment>> columnSegments, final Map<String, String> columnTableNames) {
+        List<SubstitutableColumnNameToken> result = new LinkedList<>();
```

Comment:
@cheese8 If you don't have to use List for operation, the Collection type is better.